Docs: Font additions, style updates, and icon implementation

Shrinks99 opened the pull request:
With this PR the docs styling is largely where I'd like it to be for now  Neat!

## Screenshots

Footer links!

<img width="773" alt="Screenshot 2023-04-06 at 11 47 22 PM" src="https://user-images.githubusercontent.com/5672810/230536984-af13ecda-dd96-49be-a99c-ea30f0c9450c.png">


## New Page Styling

Goal for this was to improve the hierarchy of titles — I generally dislike thin fonts, hard to scan through documents quickly — and code blocks, as well as implementing the colour scheme I've been working on to differentiate our docs / tool branding with different hues for each of our different tools as mkdocs eventually gets extended and standardized to the other pieces of software we have.

![Screenshot 2023-04-07 at 00-05-31 Local Deployment - Browsertrix Cloud Docs](https://user-images.githubusercontent.com/5672810/230539112-e9289e70-63e9-4e80-9719-aa51554117d5.png)

Kind of like _this!_ 👀 .... But that's for another repo! :)

![Screenshot 2023-04-07 at 03-58-40 Embedding ReplayWeb page - ReplayWeb Page Docs](https://user-images.githubusercontent.com/5672810/230568093-1f21cd4a-4760-44ba-a069-9f67079c6f74.png)

## Changes
- Updates background & code block colours
- Adds footer links!
- Updates header styles
- Changes body font to Recursive with appropriate variable axis settings (not monospaced for body text, 0.51 for headers)
  - We self host it now too!
- Changes header text color
- Reverts body text color to default (it was the dark green before, meant to only use that for large titles but it wasn't obvious)
- Adds only the icons we use to the repo along with instructions for adding more.
- Updates admonition icons (see screenshot above)
  - Changed type from "tip" to "info" where they're used in the deployment section
  - Switched all the other icon types over to Bootstrap's icons in the config
- Adds docs license info to both the footer & repo readme
  - Removes the dual-licensing bit because we're not doing that?

Shrinks99 committed:
Add stylesheet & mkdocs features

- Adds a custom stylesheet & brand colours
- Adds Recursive as the code font
- Adds repo info to the nav bar
- Adds auto tracking ID links for deep linking to sections as users scroll the page
- Index pages are now a part of their section as determined by their H1
- Removes mkdocs info from future footer

Shrinks99 committed:
Reorganize content

- Renames "Dev" to "Develop" for improved navigation labels
- Deployment docs are now located under a larger "Development" section (fewer nav bar choices & realistically I think anyone who wants to do one is going to be referring to the other)
- Adds links to tools the first time they're mentioned
- Rewords part of the homepage
- Hides section navigation on the homepage (now we don't have a blank section nav bar!
- Adds some syntax highlighting
- Removes some manual word wrapping — this was done very rarely / inconsistently

Shrinks99 committed:
Rename "Developer Docs" index page

- Better title for sidebar

Shrinks99 committed:
Update docs.md

- Adds links to tools
- Adds future docs style guide section
- Updates name and makes it an H1

Shrinks99 committed:
Moves deployment docs back to the root docs directory

- Replaces hyphens on the homepage with em dashes

Shrinks99 committed:
Adds bootstrap icons to the docs repo

Shrinks99 committed:
Adds footer license & links

- Updates license section in readme clarifying docs licensing

Shrinks99 committed:
update admonition icons

Shrinks99 committed:
merge main into update

Shrinks99 committed:
Typography updates

- Sets Recursive as the main typeface for code and text!
- Adjusts variable axes and sets stylistic alternates accordingly.
- Self hosts the font

Shrinks99 committed:
Adds the bootstrap icon library to the docs dir

Shrinks99 committed:
Merge branch 'main' into docs-fonts&icons

Shrinks99 committed:
Updates main & code block background colors

Shrinks99 committed:
removes icons

Shrinks99 commented:
Keeping Ilya's reasonable request to not double the repo's size with a bunch of icons — most of which aren't used at this point and probably never will be — in mind... it seems the best way may be to only copy over the icons we want to use to the `overrides/.icons/bootstrap` directory.  We can't effectively use a CDN (as far as I'm aware) because mkdocs requires us to refer to a local file path for icons defined in the mkdocs.yml file.

If there's a clever way to do this, I'd love to know!

Shrinks99 committed:
Adds the specific icons currently required

- Updates writing docs page regarding adding icons

tw4l reviewed (commented):
Overall looks great! Only thing I noticed in WAVE is that there are a few places where the color contrast is below what WCAG suggests, e.g.:

<img width="631" alt="Screen Shot 2023-04-11 at 11 26 33 AM" src="https://user-images.githubusercontent.com/6758804/231212537-7a1620ad-5d1e-4fa2-8dc0-334410267760.png">

<img width="453" alt="Screen Shot 2023-04-11 at 11 26 50 AM" src="https://user-images.githubusercontent.com/6758804/231212560-f31691b8-4869-4e81-a7b8-206d8c4c9d3d.png">

Shrinks99 commented:
@tw4l WCAG2's contrast algorithm notably STINKS!  It is wholly unrepresentative of how humans interpret colours, not accounting for hue & other perceptual phenomena when calculating a contrast score.  For more information, check out [this article with nice examples](https://blog.datawrapper.de/color-contrast-check-data-vis-wcag-apca/)!

The nav is a great example of this where sRGB white text on sRGB #008873 bg fails WCAG2 but passes APCA for body text — and rightfully so?  Looks plenty legible to me.

That said, your point is still valid and [through APCA it only gets a "fluent text okay"](https://www.myndex.com/APCA/?BG=f4f0e7&TXT=008873&DEV=G4g&BUF=A22) which isn't ideal considering the weight used here (about 200 off from what the calculator considers accessible).  I'm glad you brought this up because when I generated this palette I didn't _fully_ validate it, looks like I need to generate a few middle ground colour values between the light and dark swatches, will probably also lighten the background some to bump the score up, ideally past Lc 70.

As for the nav link opacity reduction, I'll see what I can do... This is kind of a Material for MKDocs accessibility issue and a good example of why I prefer to not use opacity for hover effects!

tw4l commented:
> WCAG2's contrast algorithm notably STINKS! It is wholly unrepresentative of how humans interpret colours, not accounting for hue & other perceptual phenomena when calculating a contrast score. For more information, check out [this article with nice examples](https://blog.datawrapper.de/color-contrast-check-data-vis-wcag-apca/)!

This is great to know, thank you!

> As for the nav link opacity reduction, I'll see what I can do... This is kind of a Material for MKDocs accessibility issue and a good example of why I prefer to not use opacity for hover effects!

Makes sense!

Everything else looks good so going to go ahead and approve so you can make the changes as you see fit and merge when ready.

Shrinks99 committed:
Link accessibility improvements

- Nav bar text is now 20% higher opacity, hover state also differentiated with weight
- In-body links are now underlined
- Lightened BG colour and darkened link colour — now achieves an APCA score of 84!

ikreymer reviewed (approved):
Nice, the non-white background is good and new icons. Could go either way on font here ;)
